build/scripts/script.js

```diff
@@ -591,6 +579,113 @@ function updateSearchResults(value) {
   }
 }
 
+/**
+ * Add missing top sections headers (h1) in search results.
+ * Search result can be h1, h2 or h3, but the search algorithm can
+ * find a match for a h3 section and not for the h1.
+ * To not display h3 orphan in the tree view, we manually to
+ * the search result the h1 that is a parent to that h3.
+ * @param {array} searchResults
+ * @returns
+ */
+function addMissingTopSections(searchResults) {
+  const h1Sections = {};
+  const missingH1 = [];
+  for (result of searchResults) {
+    if (!h1Sections[result.doc.h1]) {
```

Comment:
Does it still work if multiple h1 in different pages have the same name.

Also, theoretically, a search result may not have any `h1`, no?